### PR TITLE
Dropdown Refactor

### PR DESCRIPTION
**Pull request by emplums**
This PR improves the dropdown component as discussed in #370!

- Added direction prop 🎉 
- Made everything styled-components-ized
- Fixed some visual bugs
- Turned all the sub components into exported nested components

Did not do: auto-closing of the dropdown when user clicks out of space. Right now it just has the default behavior of a details element, which does not close the content when clicking out of the space. Should be easy to add either in a container component or in this component as a follow up!

Closes: #370 
#### If development process was changed
Description of changes

- [ ] Updated README

#### Merge checklist
- [ ] Changed base branch to release branch
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

### CODE REVIEW

File: pages/components/docs/Dropdown.md [hunk 1]
```diff
@@ -3,26 +3,44 @@
...
 
 ## Component props
 
+
+#### Dropdown.Menu
 | Name | Type | Default | Description |
 | :- | :- | :-: | :- |
-| scheme | String | | Button scheme used to style the component, can be one of `danger`, `outline` or `primary` |
-| title | String | | Optional prop to set the text in the Dropdown button
+| direction | String | 'sw' | Sets the direction of the dropdown menu. |
```
Comment (colebemis) [inline pages/components/docs/Dropdown.md]: Can't wait until this documentation is automated! 😄

Comment (emplums) [reply pages/components/docs/Dropdown.md]: Same 😭

File: src/Dropdown.js [hunk 2]
```diff
@@ -1,60 +1,132 @@
...
+  ${COMMON}
+`
 
-Dropdown.defaultProps = {
+Dropdown.Button = styled(DropdownButton)(COMMON)
+Dropdown.Menu = DropdownMenu
+Dropdown.Item = DropdownItem
+
+Dropdown.Menu.propTypes = {
+  direction: PropTypes.oneOf(['ne', 'e', 'se', 's', 'sw', 'w']),
```
Comment (cheshire137) [inline src/Dropdown.js]: No `'n'`?

File: src/Dropdown.js [hunk 3]
```diff
@@ -1,60 +1,132 @@
...
 Dropdown.propTypes = {
   children: PropTypes.node,
-  scheme: Button.propTypes.scheme,
-  title: PropTypes.string,
+  ...COMMON.propTypes
+}
+
+Dropdown.Button.defaultProps = {theme}
+Dropdown.Button.propTypes = {
+  ...COMMON.propTypes
```
Comment (cheshire137) [inline src/Dropdown.js]: Does this go against the docs above?

> Dropdown.Menu, Dropdown.Button, and Dropdown.Item do not get any system props.

File: src/Dropdown.js (hunk 2 above)
Comment (emplums) [reply src/Dropdown.js]: Nope! We don't support 'n' yet!

File: src/Dropdown.js (hunk 3 above)
Comment (emplums) [reply src/Dropdown.js]: Oh thank you, good catch! I added that in last minute

**Pull request merged**